### PR TITLE
Set `STYX_TRIGGER_ID`

### PR DESCRIPTION
**Pull request by kanterov**
Need this feature a lot!

I'm not sure this is an optimal approach, but seems to work. Of course, I didn't test it anyhow except unit tests, because I have no idea how to do it.

### CODE REVIEW

File: styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/DockerRunnerHandler.java [hunk 1]
```diff
@@ -141,7 +141,8 @@ private String dockerRunnerStart(RunState state) throws IOException {
         dockerImage,
         ImmutableList.copyOf(command),
         executionDescription.dockerTerminationLogging(),
-        executionDescription.secret());
+        executionDescription.secret(),
+        state.data().triggerId());
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/DockerRunnerHandler.java]: `triggerId` is probably deprecated as noted by comment. `TriggerUtil.triggerId(trigger)` feels better. @rouzwawi ?

File: styx-scheduler-service/src/test/java/com/spotify/styx/SystemTest.java [hunk 2]
```diff
@@ -338,7 +343,7 @@ public void runsDockerImageWithArgsTemplate() throws Exception {
     WorkflowInstance workflowInstance = dockerRuns.get(0)._1;
     RunSpec runSpec = dockerRuns.get(0)._2;
     assertThat(workflowInstance.workflowId(), is(HOURLY_WORKFLOW.id()));
-    assertThat(runSpec, is(RunSpec.simple("busybox", "--hour", "2016-03-14T15")));
+    assertThat(runSpec, is(naturalRunSpec("busybox", ImmutableList.of("--hour", "2016-03-14T15"), "natural-trigger")));
```
Comment (honnix) [inline styx-scheduler-service/src/test/java/com/spotify/styx/SystemTest.java]: `TriggerUtil.NATURAL_TRIGGER_ID`

File: styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/DockerRunnerHandler.java (hunk 1 above)
Comment (rouzwawi) [inline styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/DockerRunnerHandler.java]: Yes

**Comment by rouzwawi** [comment]
LGTM overall. Just change to use `StadeData.trigger()` with `TeiggerUtil`. I believe that could also make the trigger id mandatory in `RunSpec`. At least that's how it should be since we should not have any executions without a trigger.

**Comment by kanterov** [comment]
@rouzwawi @honnix thanks for review, done.

@rouzwawi `triggerId` is still optional because `trigger` in `runState` is optional.

File: styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/DockerRunnerHandler.java [hunk 3]
```diff
@@ -39,6 +39,8 @@
 import java.util.List;
 import java.util.Optional;
 import java.util.concurrent.ExecutorService;
+
+import com.spotify.styx.util.TriggerUtil;
```
Comment (honnix) [inline styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/DockerRunnerHandler.java]: organize import? it shall be before `java.*` somewhere.

File: styx-scheduler-service/src/test/java/com/spotify/styx/SystemTest.java [hunk 4]
```diff
@@ -51,6 +52,8 @@
 import java.time.temporal.ChronoUnit;
 import java.util.Arrays;
 import java.util.Optional;
+
+import com.spotify.styx.util.TriggerUtil;
```
Comment (honnix) [inline styx-scheduler-service/src/test/java/com/spotify/styx/SystemTest.java]: same here.

**Comment by honnix** [comment]
Minor issue. Otherwise LGTM. 👍

**Pull request merged**